src/elements/Modal/RequestIRECsModal.tsx

```diff
@@ -0,0 +1,172 @@
+import { Modal, Button, Dropdown, DropdownButton, MenuItem } from 'react-bootstrap';
+import './Modal.scss';
+import '../PageButton/PageButton.scss';
+import moment from 'moment';
+
+import { Configuration } from 'ew-utils-general-lib';
+import { CertificateLogic } from 'ew-origin-lib';
+import { ProducingAsset } from 'ew-asset-registry-lib';
+
+import { showNotification, NotificationType } from '../../utils/notifications';
+import DatePicker from 'react-date-picker';
+
+interface IRequestIRECsModalProps {
+    conf: Configuration.Entity;
+    producingAsset: ProducingAsset.Entity;
+    showModal: boolean;
+    callback: () => void;
+}
+
+interface IRequestIRECsModalState {
+    show: boolean;
+    fromDate: Date;
+    toDate: Date;
+    reads: any[];
+}
+
+export class RequestIRECsModal extends React.Component<IRequestIRECsModalProps, IRequestIRECsModalState> {
+    constructor(props, context) {
+        super(props, context);
+
+        this.handleClose = this.handleClose.bind(this);
+        this.requestIRECs = this.requestIRECs.bind(this);
+        this.isFormValid = this.isFormValid.bind(this);
+        this.handleToDateChange = this.handleToDateChange.bind(this);
+
+        this.state = {
+            show: props.showModal,
+            fromDate: moment().toDate(),
+            toDate: this.setMaxTimeInDay(moment().add('1', 'month').toDate()),
+            reads: []
+        };
+    }
+
+    componentWillReceiveProps(newProps: IRequestIRECsModalProps) {
+        this.setState({
+            show: newProps.showModal
+        });
+    }
+
+    async componentDidUpdate(prevProps : IRequestIRECsModalProps) {
+        if (this.props.producingAsset && (this.props.producingAsset !== prevProps.producingAsset)) {
+            const reads = await this.props.producingAsset.getSmartMeterReads();
+
+            const certificateLogic : CertificateLogic = this.props.conf.blockchainProperties.certificateLogicInstance;
+
+            const requestedSMReadsLength = Number(await certificateLogic.getAssetRequestedCertsForSMReadsLength(Number(this.props.producingAsset.id)));
+
+            const fromDate = moment.unix(reads[requestedSMReadsLength].timestamp).toDate();
+
+            this.setState({
+                reads,
+                fromDate
+            });
+        }
+    }
+
+    async requestIRECs() {
+        const certificateLogic : CertificateLogic = this.props.conf.blockchainProperties.certificateLogicInstance;
+
+        const readsInRange = this.getReadsInTimeRange(this.state.fromDate, this.state.toDate);
+        const lastReadIndex = this.state.reads.indexOf(readsInRange[readsInRange.length - 1]);
+
+        await certificateLogic.requestCertificates(Number(this.props.producingAsset.id), lastReadIndex);
+
+        const energy = this.getEnergy(this.state.fromDate, this.state.toDate) / 1000;
+
+        showNotification(`Certificates for ${energy} kWh requested.`, NotificationType.Success);
+
+        this.props.callback();
```

Comment:
`this.props.callback()` would work better if put directly in the `handleClose()` function.